runtime/src/main/java/com/dylibso/chicory/runtime/Module.java

```diff
@@ -226,11 +226,9 @@ protected Instance instantiate(
 
         Memory memory = null;
         if (module.memorySection() != null) {
```

Comment:
When we cleanup the APIs, we should consider changing all of these nullables to `Optional`

Reply:
Sadly `Optional`s are a leaky abstraction in Java, I would be happy to discuss this change, I might be overthinking something.